### PR TITLE
feat: retry on network errors

### PR DESCRIPTION
**Pull request by nd0ut**

### CODE REVIEW

File: packages/upload-client/README.md [hunk 1]
```diff
@@ -364,6 +364,19 @@ Sets the maximum number of attempts to retry throttled requests.
 
 Defaults to `1`.
 
+#### `retryNetworkErrorMaxTimes: number`
```
Comment (nd0ut) [inline packages/upload-client/README.md:367]: @KamillaGarifullina 
Check this text please 🙏

File: packages/upload-client/README.md [hunk 2]
```diff
@@ -364,6 +364,19 @@ Sets the maximum number of attempts to retry throttled requests.
...
+
+Sets the maximum number of attempts to retry requests failed with network error.
```
Comment (KamillaGarifullina) [inline packages/upload-client/README.md:369]: "failed with network" -> "that failed with a network"

File: packages/upload-client/README.md [hunk 3]
```diff
@@ -364,6 +364,19 @@ Sets the maximum number of attempts to retry throttled requests.
...
+
+Defaults to `3`.
+
+Delay between attempts equals attempt number, i.e.
```
Comment (KamillaGarifullina) [inline packages/upload-client/README.md:373]: "Delay" -> "The delay"

**Review by KamillaGarifullina — APPROVED** [review]

File: packages/upload-client/src/defaultSettings.ts [hunk 4]
```diff
@@ -9,11 +9,11 @@ const defaultSettings: DefaultSettings = {
   baseURL: 'https://upload.uploadcare.com',
   maxContentLength: 50 * 1024 * 1024, // 50 MB
   retryThrottledRequestMaxTimes: 1,
+  retryNetworkErrorMaxTimes: 3,
   multipartMinFileSize: 25 * 1024 * 1024, // 25 MB
   multipartChunkSize: 5 * 1024 * 1024, // 5 MB
   multipartMinLastPartSize: 1024 * 1024, // 1MB
   maxConcurrentRequests: 4,
```
Comment (FilippovAM) [inline packages/upload-client/src/defaultSettings.ts:16]: Can somebody use `multipartMaxAttempts`?

**Review by FilippovAM — APPROVED** [review]

File: packages/upload-client/src/defaultSettings.ts (hunk 4 above)
Comment (nd0ut) [reply packages/upload-client/src/defaultSettings.ts:16]: @FilippovAM Yep. This is breaking change and we will release major version, see commit https://github.com/uploadcare/uploadcare-js-api-clients/pull/424/commits/178436e83c956a2b428e11bc4d76af27b627bcdb

**Pull request merged**